test/unit/helpers/samples_helper_test.rb

```diff
@@ -82,4 +82,39 @@ class SamplesHelperTest < ActionView::TestCase
       refute attempt_to_show_extract_samples_button?(data_file, data_file.latest_version)
     end
   end
+
+  test 'cv attribute display with free text' do
+    person = FactoryBot.create(:person)
+    User.with_current_user(person.user) do
+      attribute = FactoryBot.create(:apples_controlled_vocab_attribute,
+                                    sample_type: FactoryBot.create(:simple_sample_type),
+                                    allow_cv_free_text:true)
+      existing_term = attribute.sample_controlled_vocab.sample_controlled_vocab_terms.first.label
+      free_text = 'new term'
+
+      [existing_term, free_text].each do |value|
+        assert_nothing_raised do
+          display_attribute_value(value, attribute)
```

Comment:
would be good if this also checked the return value. Even though an exception is no longer raised, it could be returning nil or something else incorrect instead.